### PR TITLE
kademlia: EachNeighbor funcs

### PR DESCRIPTION
**Pull request by esadakar**

### CODE REVIEW

File: pkg/kademlia/kademlia_test.go [hunk 1]
```diff
@@ -146,6 +146,83 @@ func TestNeighborhoodDepth(t *testing.T) {
...
+		addr := test.RandomAddressAt(base, i)
+		peers = append(peers, addr)
+	}
+
+	add(t, signer, kad, ab, peers, 0, 15)
+	waitCounter(t, &conns, 15)
+
+	var depth uint8 = 15
+
+	kad.EachNeighbor(func(adr swarm.Address, po uint8) (stop, jumpToNext bool, err error) {
```
Comment (janos) [inline pkg/kademlia/kademlia_test.go:200]: Check for returned error.

File: pkg/kademlia/kademlia_test.go [hunk 2]
```diff
@@ -146,6 +146,83 @@ func TestNeighborhoodDepth(t *testing.T) {
...
+		}
+		return false, false, nil
+	})
+
+	if depth < kad.NeighborhoodDepth() {
+		t.Fatalf("iterator passed in an out of depth peer")
+	}
+
+	depth = 15
+	kad.EachNeighborRev(func(adr swarm.Address, po uint8) (stop, jumpToNext bool, err error) {
```
Comment (janos) [inline pkg/kademlia/kademlia_test.go:213]: Check for returned error.

File: pkg/kademlia/kademlia.go [hunk 3]
```diff
@@ -888,6 +888,34 @@ func (k *Kad) ClosestPeer(addr swarm.Address, skipPeers ...swarm.Address) (swarm
 	return closest, nil
 }
 
+// IsWithinDepth returns if an address is within the neighborhood depth of a node
```
Comment (janos) [inline pkg/kademlia/kademlia.go:891]: Use `.` at the end of the comment to represent the whole sentence. https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

This goes for other comments in this PR.

**Review by janos — APPROVED** [review]
Thanks @esadakar! LGTM.

File: pkg/kademlia/kademlia_test.go [hunk 4]
```diff
@@ -146,6 +146,89 @@ func TestNeighborhoodDepth(t *testing.T) {
...
+	}
+
+	add(t, signer, kad, ab, peers, 0, 15)
+	waitCounter(t, &conns, 15)
+
+	if kad.IsWithinDepth(peers[0]) {
+		t.Fatalf("peer should NOT be in neignborhood")
+	}
+
+	if !kad.IsWithinDepth(peers[14]) {
```
Comment (zelig) [inline pkg/kademlia/kademlia_test.go:173]: a more exact test would be better

File: pkg/kademlia/kademlia_test.go [hunk 5]
```diff
@@ -146,6 +146,89 @@ func TestNeighborhoodDepth(t *testing.T) {
...
+			depth = po
+		}
+		return false, false, nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if depth < kad.NeighborhoodDepth() {
+		t.Fatalf("iterator passed in an out of depth peer")
```
Comment (zelig) [inline pkg/kademlia/kademlia_test.go:212]: incorrect depth argument pass to iterator function: expected >= %d (neighbourhood depth), got %d

**Review by zelig — APPROVED** [review]

**Pull request merged**